### ISSUE TITLE
Cloudflare DNS Lets Encrypt 

### ISSUE THREAD

**Issue by ikifar2012** [jc21/nginx-proxy-manager#120]
Please add the ability to do a DNS challenge for lets encrypt

**Comment by rudyberkvens** [comment]
I need this also. My ISP blocks port 80

**Comment by ikifar2012** [comment]
What ISP is that?

**Comment by rudyberkvens** [comment]
SuddenLink
They also block 25 as well as others. They are not willing to unblock for me.

Thanks,

**Comment by jc21** [comment]
Time for a new ISP ;) Do they block 443 as well?

Check out #85 - it needs testing, you can pull the docker image for that PR and try

**Comment by rudyberkvens** [comment]
Seems to be very common among residential ISPs. I wish there were other for me to pic from. No fortunately they leave 443 alone.
For now I’m manually requesting from https://www.sslforfree.com/ and importing into the proxy manager UI. It woks but not ideal.

**Comment by Deanosim** [comment]
> Time for a new ISP ;) Do they block 443 as well?
> 
> Check out #85 - it needs testing, you can pull the docker image for that PR and try

How do I go about testing this? I've pulled it but I don't know where to enter the the info for setting up dns requests via the cloudflare plugin.

**Comment by jc21** [comment]
According to the guy who wrote the PR, once you've pulled it you need to have your dns settings done and then go and add request a new LE cert from within the application. There's no new UI options.

**Comment by jc21** [comment]
Merged #85

**Comment by ikifar2012** [comment]
hey @jc21 hpw do I use the dns challange

**Comment by jc21** [comment]
The patch only tells letsencypt to search your dns settings for some authorization instead of inside the project via http. I personally don't use it but someone wanted it, made a PR, was told to test it and I didn't hear from them again so I merged it because it doesn't affect any pre-existing behavior.

**Comment by ikifar2012** [comment]
> The patch only tells letsencypt to search your dns settings for some authorization instead of inside the project via http. I personally don't use it but someone wanted it, made a PR, was told to test it and I didn't hear from them again so I merged it because it doesn't affect any pre-existing behavior.

In order for Cloudflare to work you need a special package for certbot and it requires a cloudflare global API key

**Comment by OhHeyAlan** [comment]
@ikifar2012 I’ll work on this tonight and try to submit a PR for it. Currently I use Certbot to generate a wildcard and install it as a custom cert.

**Comment by burkasaurusrex** [comment]
FYI, it looks like Cloudflare posted a FAQ on this: https://support.cloudflare.com/hc/en-us/articles/214820528-Validating-a-Let-s-Encrypt-Certificate-on-a-Site-Already-Active-on-Cloudflare

**Comment by dcplaya** [comment]
Is this working? I didnt see anything in the github wiki about it. I would love to use this but I need to have LetsEncrypt to auth with Cloudflare's DNS before I can switch over to it.

**Comment by rudyberkvens** [comment]
I switched to using Let's Encrypt and NGINX Home Assistant SSL proxy add-ins in Home Assistant. These two together are working fine for me. Of course this wont help you if you are using this for something other than HA.

**Comment by dcplaya** [comment]
I use HA but I also have about 15 other services running through my proxy. But I currently am unable to use this anyway, most of my services' ports are only opened to localhost and as far as I can tell, there is no way to have Nginx look at localhost of the parent machine and not the docker instance. So for now, implementing this is on hold.
